### ISSUE TITLE
How to disable build-in commands

### ISSUE THREAD

**Issue by crile** [zetamatta/nyagos#326]
Hi, 

Really nice program. However, I encounter few issues.
I don't like the "del" and "rmdir" alias. I am used to type "rmdir /s /q" for remove complete directory structure, but with the build-in command, I can't. I don't find the way to remove directory recursively in nyagos.

Another issue, the ~ substitution, it really useful but I use "pscp" command to copy file on remote Linux machine. And it fails in nyagos. I suppose that the ~ in the command "pscp file.txt user@machine:~/" is wrongly replaced.

So my question is, how can I disable or customize the build-in commands ?

Thanks

**Comment by zetamatta** [comment]
Thank you for opening the issue !

There are some ways. If it is the command CMD.exe has, we can use it by using aliases like this.

```
nyagos.alias.rmdir = [[cmd /c "rmdir $*"]]
```

If it is the executable which has an extension, for example `foo.exe` ,

```
nyagos.alias.foo = "foo.exe $*"
```

Writing them on `%USERPROFILE%\.nyagos` , you can use commands not nyagos' build-in version.

**Comment by crile** [comment]
Thanks, it works. 
No answer for the `~` ? :)

**Comment by zetamatta** [comment]
I am very sorry. I missed it.

1. The ~ substitution works on the tilde of the top of the word (for example ,1-space + tilde-mark or tilde-mark on the top of the line).
2. The ~ substitution does not work in the area enclosed with double-quatations.

So, I consider the tilde in `user@machine:~/` is **not** substituted to the value of the environment variable USERPROFILE, is it ?

( If you want to disable the ~ substitution **always**, I can create option to disable the substitution on the next version.  )

**Comment by zetamatta** [comment]
The **next** version of NYAGOS supports this :

![image](https://user-images.githubusercontent.com/3752189/43336043-56458e08-920b-11e8-94ed-7a9d46cb11ab.png)

**Comment by crile** [comment]
Thank you. My bad for the `~` issue. It is not linked to the nyagos.
But it is still nice to have some option to disable it.

**Comment by zetamatta** [comment]
`tilde_expansion` is implemented on 4.3.3. Can I close this issue ?

**Comment by crile** [comment]
sure you can. Thank you again
